src/libstd/ffi/c_str.rs

```diff
@@ -522,6 +522,37 @@ impl ToOwned for CStr {
     }
 }
 
+#[unstable(feature = "cstring_asref", reason = "recently added", issue = "0")]
+impl ops::Index<ops::RangeFull> for CString {
```

Comment:
Probably should implement indexing and slicing with other kind of ranges as well then. Seems strange to only support slicing by RangeFull.